Makefile

```diff
@@ -43,32 +43,45 @@ fpm: extract npm_verify
 		--exclude opt/$(PKGNAME)/$(PKGNAME)-$(PKGVER) \
 		-n $(PKGNAME) -v $(PKGVER) -C target
 
+patch: $(PKGDIRNAME)
+	@cd $(PKGDIRNAME) && find ../patches -type f -name '*.patch' -print0 | sort -z | xargs -t -0 -n 1 patch --verbose -p1 -i
+
 npm_download: extract
 	@cd $(PKGDIRNAME) && npm i --production
 
 npm_verify: npm_download
 	cat $(NPMS) | sha256sum -c
 
-regenerate_sums: $(PKGTARBALL)
+regenerate_sums: npm_download
 	@echo Generating NEW checksums...
 	find $(PKGDIRNAME)/node_modules/ -type f -exec sha256sum {} \; > npm_modules.sha256sum
 
 extract: $(PKGDIRNAME)
 $(PKGDIRNAME): verify
-	tar xvzf $(PKGTARBALL)
+	tar xvzf $(PKGARCHIVE) || unzip $(PKGARCHIVE)
 
-download: $(PKGTARBALL)
-$(PKGTARBALL):
+download: $(PKGARCHIVE)
+$(PKGARCHIVE):
 	@echo Getting package release $(PKGVER)...
-	curl -# -L -O $(PKGPATH)$(PKGTARBALL)
+	curl -# -L -O $(PKGPATH)$(PKGARCHIVE)
+
+verify: $(PKGARCHIVE)
+	@echo Verifying package checksum...
+	echo "$(PKGSHA256) $(PKGARCHIVE)" | sha256sum -c
 
-verify: $(PKGTARBALL)
-	@echo Verifying tarball checksum...
-	echo "$(PKGSHA256) $(PKGTARBALL)" | sha256sum -c
+setup:
+	sudo --validate
+	sudo yum update -y
+	test -e /etc/yum.repos.d/nodesource-el7.repo || curl -sL https://rpm.nodesource.com/setup_10.x | sudo bash -
+	sudo yum install -y git unzip rpm-build nodejs gcc gcc-c++ patch autoconf automake bison libffi-devel libtool readline-devel sqlite-devel zlib-devel openssl-devel
+	gpg2 --keyserver hkp://pool.sks-keyservers.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+	test -e ~/.rvm/scripts/rvm || curl -sSL https://get.rvm.io | bash -s stable
+	~/.rvm/bin/rvm install 2.7.2
```

Comment:
Should this version number be a variable at the top?

Reply:
Good idea, I'll do that.

Reply:
I've implemented your suggestion in 42fa0e560d437768800c711c10412bb37634c98a